### PR TITLE
Fix kernel argument passing for aggregates

### PR DESCRIPTION
**Pull request by linehill**
Fix kernel argument passing for aggregates

Fix sizes of aggregate arguments which were not computed correctly causing kernel arguments to be copied partially or copy more bytes than needed to device. The issues found and fixed in the argument size computation logic:

* Array lengths were not extracted from the SPIR-V properly causing random argument sizes.
* The argument size computation logic for structures/arrays did not account alignment requirements of its members/elements.

### CODE REVIEW

**Review by pvelesko — APPROVED** [review]
Looks good but can you provide some background on how these issues were discovered? Were no unit tests affected by this? 

Will merge after after I merge the CI upgrades.

**Comment by linehill** [comment]
> Looks good but can you provide some background on how these issues were discovered? 

It's already recognized issue (#279).

> Were no unit tests affected by this?

By a glance they don't seem to test kernel aggregate arguments.

**Pull request merged**
